server/services/permissionsService.test.ts

```diff
@@ -0,0 +1,785 @@
+
+        const permissions = await service.getOverviewPermissions(user, PrisonerMockDataA)
+
+        expect(permissions.category).toEqual({
+          view: false,
+          edit: true,
+        })
+      })
+    })
+
+    describe('calculateReleaseDates', () => {
+      it('should return edit false if user doesnt have appropriate role', async () => {
+        const permissions = await service.getOverviewPermissions(prisonUserMock, PrisonerMockDataA)
+
+        expect(permissions.category).toEqual({
+          view: false,
+          edit: false,
+        })
+      })
+
+      it('should return edit true if user has ReleaseDatesCalculator role', async () => {
+        const user = { ...prisonUserMock, userRoles: [Role.ReleaseDatesCalculator] }
+
+        const permissions = await service.getOverviewPermissions(user, PrisonerMockDataA)
+
+        expect(permissions.calculateReleaseDates).toEqual({
+          view: false,
+          edit: true,
+        })
+      })
+    })
+
+    describe('caseNotes', () => {
+      it('should return edit true if prisoner is in users case loads', async () => {
+        const permissions = await service.getOverviewPermissions(prisonUserMock, PrisonerMockDataA)
+
+        expect(permissions.caseNotes).toEqual({
+          view: false,
+          edit: true,
+        })
+      })
+
+      it('should return edit false if prisoner is not in users case loads', async () => {
+        const user = {
+          ...prisonUserMock,
+          caseLoads: [{ caseLoadId: 'LEI', currentlyActive: true } as CaseLoad],
+          userRoles: [Role.GlobalSearch],
+        }
+
+        const permissions = await service.getOverviewPermissions(user, PrisonerMockDataA)
+
+        expect(permissions.caseNotes).toEqual({
+          view: false,
+          edit: false,
+        })
+      })
+
+      it('should return edit true if user has GlobalSearch and PomUser roles', async () => {
+        const user = {
+          ...prisonUserMock,
+          userRoles: [Role.GlobalSearch, Role.PomUser],
+          caseLoads: [{ caseLoadId: 'LEI', currentlyActive: true } as CaseLoad],
+        }
+
+        const permissions = await service.getOverviewPermissions(user, PrisonerMockDataA)
+
+        expect(permissions.caseNotes).toEqual({
+          view: false,
+          edit: true,
+        })
+      })
+
+      it.each([Role.GlobalSearch, Role.PomUser])('should return edit false if user has only %s role', async role => {
+        const user = {
+          ...prisonUserMock,
+          userRoles: [role],
+          caseLoads: [{ caseLoadId: 'LEI', currentlyActive: true } as CaseLoad],
+        }
+
+        const permissions = await service.getOverviewPermissions(user, PrisonerMockDataA)
+
+        expect(permissions.caseNotes).toEqual({
+          view: false,
+          edit: false,
+        })
+      })
+
+      it('should return edit true if prisoner is out and user has inactive bookings role', async () => {
```

Comment:
I think we're missing a check of `should return edit false if prisoner is out` (`and user does not have inactive bookings role`)

Reply:
Same again for transfer?

I know they're effectively just like any other case load, so if you don't have them you haven't got access, but it might be worth explicitly calling out in the tests?

Reply:
If the prisoner is out and the user doesn't have that role they won't get any permissions, it will bail out at the access section  tested [here](https://github.com/ministryofjustice/hmpps-prisoner-profile/pull/658/files#diff-95688007aec6993ad05ed72087f054153d07188c542f7d7af6f5557d1a305902R80)

Reply:
Agreed, but I shouldn't this permissionsService be tested independently of the permissionsGuard? Just thinking since in theory this could be ported over to a separate service sometime in the far future and it'd be good to have a complete set of tests

Reply:
Ok so that means I need to extract all of this and call separately